keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ArtifactInterfaces.kt

```diff
@@ -52,7 +52,27 @@ interface ArtifactReferenceProvider {
     }
 }
 
-interface ComputeResourceSpec : ResourceSpec, VersionedArtifactProvider, ArtifactReferenceProvider
+interface ComputeResourceSpec : ResourceSpec, VersionedArtifactProvider, ArtifactReferenceProvider {
+  val clusterHealth: ClusterHealth?
+}
+
+/**
+ * Contains user-configured health information about a server group
+ * [ignoreHealthForDeployments] if true, considers deployments healthy when instances are not in a
+ *  [InstanceCounts] down, out of service, or starting status. This is the setting for apps which
+ *  "only consider provider health"
+ * [healthyPercentage] the percentage of instances that need to be in the "up" status before
+ *  we consider a cluster fully deployed
+ */
+data class ClusterHealth(
+  val ignoreHealthForDeployments: Boolean = false,
```

Comment:
Could this name be clearer for users?

Reply:
+1. In fact, this is even a bit paradoxical given the name of the class. Can't a user just set `healthyPercentage` to zero if they don't care about it?